ui/hooks/flask/useTransactionInsightSnap.js

```diff
@@ -5,28 +5,33 @@ import { getPermissionSubjects } from '../../selectors';
 
 const INSIGHT_PERMISSION = 'endowment:transaction-insight';
 
-export function useTransactionInsightSnap(transaction, snapId) {
+export function useTransactionInsightSnap({ transaction, chainId, snapId }) {
   const subjects = useSelector(getPermissionSubjects);
   if (!subjects[snapId]?.permissions[INSIGHT_PERMISSION]) {
     throw new Error(
       'This snap does not have the transaction insight endowment.',
     );
   }
-  const [data, setData] = useState(null);
+  const [data, setData] = useState(undefined);
 
   useEffect(() => {
     async function fetchInsight() {
       const d = await handleSnapRequest({
-        snapId, origin: undefined, handler: 'onTransaction', request: {
-          params: [transaction],
-        }
+        snapId,
+        origin: 'test',
+        handler: 'onTransaction',
+        request: {
```

Comment:
Is the request object validated to have a method property? If not, we can remove this.

Reply:
Yes, it's expecting a proper JSON-RPC method so it's the only way to make it work